packages/organizations/origin-organization-irec-api/src/irec/irec.service.ts

```diff
@@ -318,6 +334,41 @@ export class IrecService implements IIrecService {
         });
     }
 
+    async redeem(
+        user: UserIdentifier,
+        assetId: string,
+        claimData: IClaimData
+    ): Promise<RedeemTransactionResult> {
+        const userClient = await this.getIrecClient(user);
+        const userConnectionInfo = await this.getConnectionInfo(user);
+
+        const userTradeAccount = await this.getTradeAccountCode(user);
+
+        const items = await userClient.account.getItems(userTradeAccount);
+        const item = items.find((i) => i.asset === assetId);
+
+        if (!item) {
+            throw new NotFoundException('IREC item not found');
+        }
+
+        const claimItem = new ReservationItem();
+        claimItem.code = item.code;
+        claimItem.amount = item.volume;
+
+        return userClient.redeem({
+            sender: userTradeAccount,
+            recipient: userTradeAccount,
+            approver: userConnectionInfo.userName,
+            volume: claimItem.amount,
+            items: [claimItem],
+            notes: '',
+            beneficiary: Number(claimData.beneficiary),
+            start: new Date(claimData.periodStartDate),
+            end: new Date(claimData.periodEndDate),
+            purpose: ''
```

Comment:
Seems that purpose field is required

Reply:
Any idea if this can be anything?
Or should I put something specific?

Reply:
As per Meerim's advice, put "GHG accounting".